OpenSim/Simulation/Model/ContactHalfSpace.cpp

```diff
@@ -59,4 +60,31 @@ SimTK::ContactGeometry ContactHalfSpace::createSimTKContactGeometry() const
     return SimTK::ContactGeometry::HalfSpace();
 }
 
+void ContactHalfSpace::generateDecorations(bool fixed, const ModelDisplayHints& hints,
+    const SimTK::State& s,
+    SimTK::Array_<SimTK::DecorativeGeometry>& geometry) const
+{
+    Super::generateDecorations(fixed, hints, s, geometry);
+
+    // There is no fixed geometry to generate here.
+    if (fixed) { return; }
+
+    // B: base Frame (Body or Ground)
+    // F: PhysicalFrame that this ContactGeometry is connected to
+    // P: the frame defined (relative to F) by the location and orientation
+    //    properties.
+    if (hints.get_show_contact_geometry()) {
+        const auto& X_BF = getFrame().findTransformInBaseFrame();
+        const auto& X_FP = getTransform();
+        const auto X_BP = X_BF * X_FP;
+        geometry.push_back(SimTK::DecorativeBrick(Vec3{.005, 0.5, 0.5})
```

Comment:
Great feedback, and comments @chrisdembia 
The Contact visualization pathway uses generateDecorations (which has little or no customization except maybe thru display hints) In most cases that's enough because the dimensions are finite and map to Geometry. Not in this case. I could add a half_plane_size to display_hints and it will apply to all HalfSpaces, other option is to have a property Vec3 display_size in ContactHalfSpace, or what I conjecture would work in the new visualizer, is allow the user to change the size and texture etc. in visualizer and keep track of that outside the model (after all this's a kind of appearance related change that depends on Visualizer capabilities) rather than a part of the model. Would love to hear your thoughts.